atc/exec/task_config_source.go

```diff
@@ -26,42 +28,32 @@ type TaskConfigSource interface {
 
 // StaticConfigSource represents a statically configured TaskConfig.
 type StaticConfigSource struct {
-	Plan atc.TaskPlan
+	Config *atc.TaskConfig
+	Vars   []boshtemplate.Variables
 }
 
 // FetchConfig returns the configuration.
 func (configSource StaticConfigSource) FetchConfig(lager.Logger, *worker.ArtifactRepository) (atc.TaskConfig, error) {
 	taskConfig := atc.TaskConfig{}
 
-	if configSource.Plan.Config != nil {
-		taskConfig = *configSource.Plan.Config
+	if configSource.Config != nil {
+		taskConfig = *configSource.Config
 	}
 
-	if configSource.Plan.Params == nil {
-		return taskConfig, nil
+	byteConfig, err := json.Marshal(taskConfig)
+	if err != nil {
+		return atc.TaskConfig{}, fmt.Errorf("failed to marshal task config: %s", err)
 	}
 
-	if taskConfig.Params == nil {
-		taskConfig.Params = map[string]string{}
+	// process task template, using cred manager vars
+	byteConfig, err = template.NewTemplateResolver(byteConfig, configSource.Vars).Resolve(true, true)
```

Comment:
Minor point: this marshal-resolve-unmarshal dance would also go away if we restrict `vars:` to only work with `file:`.

Reply:
agreed